core/src/main/scala/kafka/log/Log.scala

```diff
@@ -272,6 +277,10 @@ class Log(@volatile var dir: File,
     } else {
       if (!dir.getAbsolutePath.endsWith(Log.DeleteDirSuffix)) {
         recoverLog()
+        segmentWithoutTimeIndex.foreach { seg =>
+          if (seg != activeSegment)
```

Comment:
It seems we are not doing this for any old version related code so far. Maybe we don't need to do this here?